packages/jest-dom-mocks/src/document.ts

```diff
@@ -0,0 +1,42 @@
+export interface MockWindow extends Window {
+  addEventListener: jest.Mock<{}> & typeof window.addEventListener;
+  removeEventListener: jest.Mock<{}> & typeof window.removeEventListener;
+}
+
+export default class Document {
+  private isUsingMockWindow: boolean = false;
+  private window?: MockWindow;
+
+  mockWindow() {
+    window.addEventListener = jest.fn();
```

Comment:
I'd also note that most of the other stuff in this package is also mocking properties of `window`, so it seems a little disingenuous to represent whether we have mocked the two functions this file cares about as `usingMockWindow` / `mockWindow`